src/index.ts

```diff
@@ -143,6 +143,12 @@ class NotificationAPIClient implements NotificationAPIClientInterface {
       notifications: (message: WS_NotificationsResponse) => {
         const notifications = message.payload.notifications;
         this.state.lastResponseNotificationsCount = notifications.length;
+
+        if (this.elements.notificationsLoading) {
```

Comment:
Either remove loading elements in render functions or handler. 
```
  renderPreferences(preferences: Preference[]): void {
    if (!this.elements.preferencesPopup) return;

    // remove loading
    this.elements.preferencesLoading?.remove();
    this.elements.preferencesLoading = undefined;
    ```

Reply:
Good point, I'll do some refactoring then.